router.go

```diff
@@ -180,7 +180,10 @@ func (r *Router) route(src io.Reader, s3url string) (map[destination]buffer, err
 			}
 			enc = r.option.newEncoder(body)
 		}
-		enc.Encode(rec, recordBytes)
+		if err := enc.Encode(rec, recordBytes); err != nil {
+			log.Println("[warn] failed to encode", err)
```

Comment:
Outputting errored record is useful for debugging.

```suggestion
			log.Println("[warn] failed to encode %s record: %v", err, rec)
```

Reply:
👍